plugins/inputs/icinga2/icinga2.go

```diff
@@ -110,6 +122,66 @@ func (i *Icinga2) createHTTPClient() (*http.Client, error) {
 	return client, nil
 }
 
+func (i *Icinga2) IcingaRequest(address string) (*http.Response, error) {
```

Comment:
Do you need those functions exported? If not (i.e. only internal use in this package) please do not export them (i.e. make them start lower-case). Same for all other functions below.

Reply:
You're right, it's not needed.